### PR TITLE
Use SQL to optimize orphaned alert pruning workflow

### PR DESCRIPTION
**Pull request by connorgorman**
## Description

Use sql instead of cursors. There are 3 million alerts in this database and it times out on cursor timeout. In this cluster there are also 0 matches. Test will be written before merging

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

Will run in the cluster

### CODE REVIEW

File: central/pruning/pruning.go [hunk 1]
```diff
@@ -458,34 +458,39 @@ func (g *garbageCollectorImpl) removeOrphanedPLOP() {
...
+		alertsToResolve = append(alertsToResolve, alert.GetId())
+		return nil
+	})
+	return alertsToResolve, err
+}
+
+func (g *garbageCollectorImpl) markOrphanedAlertsAsResolved(deployments set.FrozenStringSet) {
+	alertsToResolve, err := g.getOrphanedAlerts(pruningCtx, deployments)
+	if err != nil {
+		log.Errorf("[Alert pruning] error getting orphaned alert ids: %v", err)
```
Comment (connorgorman) [inline central/pruning/pruning.go:491]: will add return below this

File: central/postgres/pruning.go [hunk 2]
```diff
@@ -35,3 +40,27 @@ func PruneClusterHealthStatuses(ctx context.Context, pool *postgres.DB) {
 		log.Errorf("failed to prune cluster health statuses: %v", err)
 	}
 }
+
+func getOrphanedAlertIDs(ctx context.Context, pool *postgres.DB) ([]string, error) {
+	var ids []string
+	rows, err := pool.Query(ctx, getAllOrphanedAlerts)
```
Comment (dashrews78) [inline central/postgres/pruning.go:46]: Do we want to give this a little more time for now?

File: central/postgres/pruning.go [hunk 3]
```diff
@@ -14,6 +16,9 @@ const (
 	pruneClusterHealthStatusesStmt = `DELETE FROM cluster_health_statuses child WHERE NOT EXISTS
 		(SELECT 1 FROM clusters parent WHERE
 		child.Id = parent.Id)`
+
+	getAllOrphanedAlerts = `SELECT id from alerts WHERE lifecyclestage = 0 and state = 0 and time < NOW() - INTERVAL '30 MINUTES' and NOT EXISTS
```
Comment (dashrews78) [inline central/postgres/pruning.go:20]: we should probably pass the orphan window into here so we only have to manage that in one place.

File: central/postgres/pruning.go (hunk 2 above)
Comment (connorgorman) [reply central/postgres/pruning.go:46]: good point, will make it 5 minutes? This one seems important to run potentially

File: central/postgres/pruning.go (hunk 3 above)
Comment (connorgorman) [reply central/postgres/pruning.go:20]: 👍

File: central/postgres/pruning.go [hunk 4]
```diff
@@ -35,3 +44,31 @@ func PruneClusterHealthStatuses(ctx context.Context, pool *postgres.DB) {
...
+	}
+	defer rows.Close()
+	for rows.Next() {
+		var id string
+		if err := rows.Scan(&id); err != nil {
+			return nil, errors.Wrap(err, "getting ids from orphaned alerts query")
+		}
+		ids = append(ids, id)
+	}
+	return ids, nil
```
Comment (dashrews78) [inline central/postgres/pruning.go:63]: We should probably return `rows.Err`.

**Review by dashrews78 — APPROVED** [review]
looks good.  let one minor comment.

**Pull request merged**